packages/preview-middleware/src/base/flex.ts

```diff
@@ -3,19 +3,60 @@ import type { ReaderCollection } from '@ui5/fs';
 import { existsSync, mkdirSync, readdirSync, unlinkSync, writeFileSync } from 'fs';
 import { join, parse } from 'path';
 
+/**
+ * Structure of a flex change.
+ */
+export interface FlexChange {
+    [key: string]: string | object | undefined;
+    changeType: string;
+    reference: string;
+    moduleName?: string;
+    content?: {
+        codeRef?: string;
+        fragmentPath?: string;
+    };
+}
+
+/**
+ * Map of change type specific correction functions.
+ */
+const moduleNameContentMap: { [key: string]: (change: FlexChange) => string } = {
+    codeExt: (change) => (change.content?.codeRef ?? '').replace('.js', ''),
+    addXML: (change: any) => change.content.fragmentPath
```

Comment:
From what I know `change.content` is always defined, even when empty. So we don't really need to check. But if you prefer to keep it, then maybe we can add the check also for the `addXML` use case?

Reply:
I'd prefer the check since it is read from the filesystem and every 'expert' could modify it.

Reply:
Good point 👍